Subject: [patch] uml free host resources associated with freed irqs.patch
From: Jeff Dike <jdike@addtoit.com>
Patch-mainline: 2.6.11-rc1

This adds a free_irq_by_irq_and_dev call which was accidentally removed when
the UML free_irq was replaced by generic code.

Signed-off-by: Jeff Dike <jdike@addtoit.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Gerd Knorr <kraxel@suse.de>
---
 arch/um/kernel/irq_user.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletion(-)

diff -puN arch/um/kernel/irq_user.c~uml-free-host-resources-associated-with-freed-irqs arch/um/kernel/irq_user.c
--- 25/arch/um/kernel/irq_user.c~uml-free-host-resources-associated-with-freed-irqs	2004-11-28 01:17:28.365776152 -0800
+++ 25-akpm/arch/um/kernel/irq_user.c	2004-11-28 01:17:28.369775544 -0800
@@ -83,8 +83,11 @@ void sigio_handler(int sig, union uml_pt
 				 * can be freed here.
 				 */
 				next = irq_fd->next;
-				if(irq_fd->freed)
+				if(irq_fd->freed){
 					free_irq(irq_fd->irq, irq_fd->id);
+					free_irq_by_irq_and_dev(irq_fd->irq,
+								irq_fd->id);
+				}
 			}
 		}
 	}
_
